tests/behat/features/api/user/search_users.feature

```diff
@@ -0,0 +1,68 @@
+@api @user
+Feature: Search users
+
+  To find users, users should be able to search all available users for specific words
+
+  Background:
+    Given there are users:
+      | name     | password | token      | id |
+      | Catrobat | 12345    | cccccccccc | 1  |
+      | User1    | vwxyz    | aaaaaaaaaa | 2  |
+      | NewUser  | 54321    | bbbbbbbbbb | 3  |
+      | Catroweb | 54321    | bbbbbbbbbb | 4  |
+    And there are followers:
+      | name     | following      |
+      | Catrobat | User1, Catroweb|
+      | NewUser  | Catrobat       |
+      | Catroweb | User1, NewUser |
+    And there are programs:
+      | id        | name            | owned by | version | private | visible |
+      | isxs-adkt | Webteam         | Catroweb | 0.8.5   | false   | true    |
+      | tvut-irkw | Catroweb        | NewUser  | 0.8.5   | false   | true    |
+    And I wait 1000 milliseconds
+
+  Scenario: Search for users with specific word in username
+
+    Given I have a parameter "query" with value "Catro"
+    And I have a request header "HTTP_ACCEPT" with value "application/json"
+    And I request "GET" "/api/users/search"
+    Then the response status code should be "200"
+    Then the response should have the users model structure
+    Then the response should contain users in the following order:
+      | Name      |
+      | Catrobat  |
+      | Catroweb  |
+
+  Scenario: Search for users with specific word in username
+
+    Given I have a parameter "query" with value "NewUser"
+    And I have a request header "HTTP_ACCEPT" with value "application/json"
+    And I request "GET" "/api/users/search"
+    Then the response status code should be "200"
+    Then the response should have the users model structure
+    Then the response should contain users in the following order:
+      | Name      |
+      | NewUser   |
+
+  Scenario: Search for users with specific word in id
+
+    Given I have a parameter "query" with value "2"
+    And I have a request header "HTTP_ACCEPT" with value "application/json"
+    And I request "GET" "/api/users/search"
+    Then the response status code should be "200"
+    Then the response should have the users model structure
+    Then the response should contain users in the following order:
+      | Name      |
+      | User1     |
+
+  Scenario: Search for users with specific word should only return users
+
+    Given I have a parameter "query" with value "Catroweb"
+    And I have a request header "HTTP_ACCEPT" with value "application/json"
+    And I request "GET" "/api/users/search"
+    Then the response status code should be "200"
+    Then the response should have the users model structure
+    Then the response should contain users in the following order:
+      | Name      |
+      | Catroweb  |
+
```

Comment:
we have many russian users, a test with russian chars would be nice :)

Reply:
Good Idea, although i don't speak Russian, i will try to write tests using Russian alphabet. @dmetzner

Reply:
Google Translate is our friend! Dosen't have to be meaningfull names just russian chars!